src/components/MailboxPicker.vue

```diff
@@ -17,37 +17,36 @@ export default {
 		Treeselect,
 	},
 	props: {
-		accountid: {
-			type: Number,
+		account: {
+			type: Object,
 			required: true,
 		},
 	},
 	computed: {
 		mailboxes() {
-			const boxes = []
-			this.getMailboxes(boxes)
-			return boxes
+			return this.getMailboxes()
 		},
 	},
 	methods: {
-		getMailboxes(root, folderid) {
-			let folders = []
-			if (!folderid) {
-				folders = this.$store.getters.getMailboxes(this.accountid)
+		getMailboxes(mailboxId) {
+			let mailboxes = []
+			const boxLevel = []
+			if (!mailboxId) {
+				mailboxes = this.$store.getters.getMailboxes(this.account.accountId)
 			} else {
-				folders = this.$store.getters.getSubMailboxes(folderid)
+				mailboxes = this.$store.getters.getSubMailboxes(mailboxId)
 			}
-			folders.forEach((folder) => {
+			mailboxes.forEach((folder) => {
 				const mailbox = {
 					id: folder.databaseId,
 					label: folder.displayName,
 				}
 				if (folder.mailboxes.length > 0) {
-					mailbox.children = []
-					this.getMailboxes(mailbox.children, mailbox.id)
+					mailbox.children = this.getMailboxes(mailbox.id)
```

Comment:
```suggestion
					mailbox.children = this.getMailboxes(mailbox.databaseId)
```

AFAIK there is no `id` on the mailbox object